lib/utils/AbstractQueryConfiguration.dart

```diff
@@ -31,23 +32,35 @@ abstract class AbstractQueryConfiguration {
 
   AbstractQueryConfiguration({
     this.language,
-    this.languages = const [],
+    this.languages,
     this.lc,
     this.cc,
     this.fields,
   }) {
     fields ??= [ProductField.ALL];
-    if ((language != null || lc != null) && languages.isNotEmpty) {
-      throw ArgumentError(
-          '[languages] cannot be used together with [language]/[lc]');
+    if (languages != null) {
+      if ((language != null || lc != null) && languages!.isNotEmpty) {
+        throw ArgumentError(
+            '[languages] cannot be used together with [language]/[lc]');
+      }
     }
   }
 
   /// Returns the corresponding API URI parameter map
   Map<String, String> getParametersMap() {
     final Map<String, String> result = {};
+    late List<OpenFoodFactsLanguage> languages;
+
+    if (language != null) {
+      languages = [language!];
+    } else if (this.languages != null && this.languages!.isNotEmpty) {
```

Comment:
`this.languages!.isNotEmpty`
Are you sure it should be illegal to pass an empty languages list?
Can't an SDK user request no languages?

Reply:
In fact, I only added this during the review reading.

True it would make sense to allow to pass an empty list to specify no language even if you have set a global value.

Reply:
Do you mean there's a discussion somewhere which prompted you to add this check?
Could you please give a link to the discussion? :)
I'm curious of the reasoning!

Reply:
Ohh, there is no other discussion, I just read the whole changed code before opening the pr, I just thought that it makes no sense passing a empty list and that we then shoud rather use the global value. But actually there is no problem to pass an empty list, because that way we create  a way to not add the lc to querie despite the global languages field is set.

Reply:
@blazern I removed the `&& this.languages!.isNotEmpty`